manticore/platforms/linux.py

```diff
@@ -96,31 +110,37 @@ def __init__(self, constraints, path="sfile", mode='rw', max_size=100, wildcard=
         :param str mode: the access permissions of the symbolic file
         :param max_size: Maximun amount of bytes of the symbolic file
         '''
-        # XXX(yan): wildcard isn't used; check callers and remove.
         assert 'r' in mode
-        if isinstance(path, str):
-            path = File(path, mode)
-        assert isinstance(path, File)
+        assert isinstance(path, str)
```

Comment:
Should we have the assert as well or just let the parent handle these (for isinstance and mode)?